CurrencyConverterControl/CurrencyConverterControl.cs

```diff
@@ -0,0 +1,251 @@
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using System.Windows.Documents;
+using System.Windows.Input;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using System.Windows.Navigation;
+using System.Windows.Shapes;
+
+namespace CurrencyConverterControl
+{
+    /// <summary>
+    /// Represents a custom WPF control for currency conversion.
+    /// This control extends the base Control class and provides functionality
+    /// for converting values between different currencies, along with a customizable
+    /// user interface.
+    /// </summary>
+    [TemplatePart(Name = "CmbDestination", Type = typeof(ComboBox))]
+    [TemplatePart(Name = "CmbSource", Type= typeof(ComboBox))]   
+    public class CurrencyConverterControl : Control, INotifyPropertyChanged
+    {
+        private static ICurrencyDataProvider CurrencyDataProvider;
+        private ComboBox _currencyDestination;
+        private ComboBox _currencySource;
+        
+        static CurrencyConverterControl()
+        {
+            DefaultStyleKeyProperty.OverrideMetadata(typeof(CurrencyConverterControl), new FrameworkPropertyMetadata(typeof(CurrencyConverterControl)));
+            CurrencyDataProvider = new CurrencyDataProvider();
+        }
+
+        #region Dependency Properties
+        /// <summary>
+        /// Gets or sets the input value for currency conversion within the CurrencyConverterControl.
+        /// </summary>
+        /// <remarks>
+        /// The InputValue property represents the value to be converted between different currencies.
+        /// It is bound to the user interface elements where users can input the amount for conversion.
+        /// </remarks>
+        [Description("Value to be converted")]
+        public double InputValue
+        {
+            get { return (double)GetValue(InputValueProperty); }
+            set { SetValue(InputValueProperty, value); }
+        }
+        public static readonly DependencyProperty InputValueProperty =
+            DependencyProperty.Register("InputValue", typeof(double), typeof(CurrencyConverterControl));
+
+        /// <summary>
+        /// Gets or sets the output value for the result of currency conversion within the CurrencyConverterControl.
+        /// </summary>
+        /// <remarks>
+        /// The OutputValue property represents the result of the currency conversion based on the provided input value.
+        /// It is bound to the user interface elements where the converted amount is displayed to the user.
+        /// </remarks>
+         [Description("Converted value")]
+        public double OutputValue
+        {
+            get { return (double)GetValue(OutputValueProperty); }
+            set { SetValue(OutputValueProperty, value); }
+        }
+
+        public static readonly DependencyProperty OutputValueProperty =
```

Comment:
you also need PropertyChangedCallback here to raise OutputValueChanged event

Reply:
Hi Irina, if l just RaisePropertychanged of INotifyPropertyChanged, i think it should be enought for this property as on the value changed of OutputValue l dont have any actions to perform as in the InputValue Prop. Is it right as a logic? 

 public double OutputValue
 {
     get { return (double)GetValue(OutputValueProperty); }
     set 
     { 
         SetValue(OutputValueProperty, value);
         RaisePropertychanged();
     }
 }

Reply:
I don't recall whether there is OutputValueChanged even in requirements. If it absent, then maybe you don't need callback here. And in general, you also won't need to call RaisePropertyChanged as well. Bindings for dependency properties should work without INotifyPropertyChanged. In this specific case you can call it, it will allow customers to subscribe on this change if they need to do something about it from code.

Reply:
> I don't recall whether there is OutputValueChanged event in requirements. If it absent, then maybe you don't need callback here. And in general, you also won't need to call RaisePropertyChanged as well. Bindings for dependency properties should work without INotifyPropertyChanged. In this specific case you can call it, it will allow customers to subscribe on this change if they need to do something about it from code.

Yeh l read on changes between INotifyPropertyChanged and PropertyChangedCallback and l removed the INotifyPropertyChanged interface as it was not needed. only by PropertyChangedCallback into the custom Libraries it is enough to cover all the actions needed. Thanks